File: src/isUnique/v1.java
package isUnique;

public class v1 {
	
	boolean isUnique(String str) {
		if(str.length() > 128) return false;
		boolean chars[] = new boolean[128];
		for(int i = 0; i < str.length(); i ++) {
			char ch = str.charAt(i);
			if(!chars[ch]) chars[ch] = true;
			else return false;
		}
		return true;
	}
	
	public static void main(String[] args) {
		v1 foo = new v1();
		
		System.out.println(foo.isUnique("Hello world!"));
		System.out.println(foo.isUnique("Liam"));
		System.out.println(foo.isUnique("Liam Patterson"));
		System.out.println(foo.isUnique("abcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()_+=[{]};:,<.>?|`~"));
	}
}

File: src/Strings/isUnique/v1.py
def isUnique(string):
    dict = {}
    for char in string:
        if char not in dict:
            dict[char] = 1
        else:
            dict[char] += 1
    for freq in dict.values():
        if freq > 1:
            return False
    return True

def prettyPrint(string):
    print("String: " + string + " " + str(isUnique(string)))

prettyPrint("Liam")
prettyPrint("aaa")
prettyPrint("  ")